File: identity_matrix_check.py
n = int(input())
mat = [list(map(int, input().split())) for _ in range(n)]
identity = all((mat[i][j] == 1 if i == j else mat[i][j] == 0) for i in range(n) for j in range(n))
print("Identity Matrix" if identity else "Not Identity")

File: identity_matrix_check.c
#include <stdio.h>
int main() {
    int n, is_identity = 1;
    scanf("%d", &n);
    int mat[n][n];
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++) {
            scanf("%d", &mat[i][j]);
            if ((i == j && mat[i][j] != 1) || (i != j && mat[i][j] != 0))
                is_identity = 0;
        }
    printf(is_identity ? "Identity Matrix\n" : "Not Identity\n");
    return 0;
}
